src/libstd/sys/unix/ext/fs.rs

```diff
@@ -178,6 +179,23 @@ impl MetadataExt for fs::Metadata {
     }
 }
 
+#[unstable(feature = "file_type_ext", reason = "recently added API")]
+pub trait FileTypeExt {
+    /// Add special unix types (block/char device, fifo and socket)
```

Comment:
This comment I believe should be placed on the trait itself. Also could you add documentation for each of these methods to say what it's doing?